### PR TITLE
Update host agent to log to a file

### PR DESCRIPTION
**Pull request by tejaskokje-mw**
Adding support to send logs to a log file. By default, the agent will log to `stdout` & `stderr`. If log file is specified, then the agent will log to that file.

Also, refactored the code for more clarity.

### CODE REVIEW

File: cmd/host-agent/main.go [hunk 1]
```diff
@@ -153,6 +165,26 @@ func app(logger *zap.Logger) *cli.App {
 				Flags:  flags,
 				Before: altsrc.InitInputSourceWithContext(flags, altsrc.NewYamlSourceFromFlagFunc("config-file")),
 				Action: func(c *cli.Context) error {
+					if cfg.Logfile != "" {
+						logger.Info("redirecting logs to logfile", zap.String("logfile", cfg.Logfile))
+						// logfile specified. Update logger to write logs to the
+						// give logfile
+						w := zapcore.AddSync(&lumberjack.Logger{
```
Comment (tejaskokje-mw) [inline cmd/host-agent/main.go:172]: lumberjack adds ability to rotate the log file.

File: cmd/host-agent/main.go [hunk 2]
```diff
@@ -45,87 +48,75 @@ func (p *program) Stop(s service.Service) error {
...
+			Name:        "logfile",
+			Usage:       "Log file to store Middleware agent logs",
+			EnvVars:     []string{"MW_LOGFILE"},
+			Destination: &cfg.Logfile,
+			DefaultText: "",
+			Value:       "",
+		}),
+
+		altsrc.NewIntFlag(&cli.IntFlag{
+			Name:        "logfile-size",
```
Comment (tejaskokje-mw) [inline cmd/host-agent/main.go:115]: This flag will allow us to rotate the logfile after the file has grown to this size.

File: configyamls/all/otel-config.yaml
Comment (bhogayatakb) [inline configyamls/all/otel-config.yaml:1]: We can skip changing this file

File: cmd/host-agent/main.go [hunk 3]
```diff
@@ -45,87 +48,75 @@ func (p *program) Stop(s service.Service) error {
...
+			Usage:       "Log file size to store Middleware agent logs. This flag only applifes if logfile flag is specified",
+			EnvVars:     []string{"MW_LOGFILE_SIZE"},
```
Comment (bhogayatakb) [inline cmd/host-agent/main.go:117]: are we setting any default value for log file size ?

**Review by bhogayatakb — APPROVED** [review]

**Pull request merged**